File: sdks/js/vercel/cases/G1.js
/**
 * G1: Basic Completion
 *
 * Tests a simple chat completion request with OpenAI SDK
 * and verifies that Sentry captures the appropriate spans and AI monitoring data.
 */

const Sentry = require("@sentry/node");
const { generateText } = require("ai");
const { openai } = require("@ai-sdk/openai");
const { getMockSentryTransport } = require("../setup");
const {
  validateFixture,
  loadFixture,
} = require("../../../../shared/test-utils/js/fixtures");

module.exports = async function () {
  console.log("    Running G1: Basic Completion");

  // Create main span for this test
  await Sentry.startSpan(
    { name: "G1-basic-completion", op: "test" },
    async () => {
      await runTest();
    }
  );

  // Flush Sentry to ensure all events are sent to transport
  await Sentry.flush(2000);

  // Small buffer to ensure transport has processed everything
  await new Promise((resolve) => setTimeout(resolve, 50));

  // Verify Sentry captured the expected data
  await assertSentryCaptured();

  console.log("    ✓ G1 completed");
};

async function runTest() {
  // Load test inputs from fixture
  const fixture = loadFixture("G1");
  const { model, system, prompt } = fixture.inputs;

  const { text } = await generateText({
    model: openai(model),
    system,
    prompt,
  });

  if (!text) {
    throw new Error("No completion returned from OpenAI");
  }

  console.log(`    Response: ${text}`);
}

async function assertSentryCaptured() {
  const transport = getMockSentryTransport();
  const spans = transport.getSpans();
  const transactions = transport.getTransactions();
  const events = transport.getEvents();

  console.log(
    `    Captured: ${spans.length} spans, ${transactions.length} transactions, ${events.length} events`
  );

  // Debug: print span ops and attributes
  console.log(`    DEBUG: Spans:`);
  spans.forEach((s, i) => {
    const dataKeys = Object.keys(s.data || {});
    console.log(`      ${i + 1}. op="${s.op}"`);
    if (dataKeys.length > 0) {
      console.log(
        `         data: ${JSON.stringify(s.data, null, 2)
          .split("\n")
          .slice(0, 10)
          .join("\n         ")}`
      );
    }
  });

  // Validate against G1 fixture
  const result = validateFixture("G1", spans, transactions, events);

  if (!result.passed) {
    console.log("    ✗ Validation failed:");
    result.errors.forEach((error) => console.log(`      - ${error}`));
    throw new Error(`Fixture validation failed:\n${result.errors.join("\n")}`);
  }

  console.log("    ✓ All fixture validations passed");
}

File: sdks/py/openai-agents/cases/G1.py
"""
G1: Basic Completion

Tests a simple agent completion request with OpenAI Agents SDK
and verifies that Sentry captures the appropriate spans and AI monitoring data.
"""

import os
import asyncio
import sentry_sdk
from agents import Agent, Runner


async def main():
    """Main test case entry point - runs the test logic only"""
    print("    Running G1: Basic Completion")

    # Run the actual test
    await run_test()

    print("    ✓ Test logic completed")


async def assert_sentry():
    """Verify Sentry captured the expected data - called after main() completes"""
    # Small buffer to ensure transport has processed everything
    await asyncio.sleep(0.1)

    await assert_sentry_captured()

    print("    ✓ G1 validation passed")


async def run_test():
    """The actual test logic"""
    # Load test inputs from fixture
    from fixtures import load_fixture

    fixture = load_fixture("G1")
    model = fixture["inputs"]["model"]
    system = fixture["inputs"]["system"]
    prompt = fixture["inputs"]["prompt"]

    # Create a simple math assistant agent
    math_agent = Agent(
        name="math_assistant",
        instructions=system,
        model=model,
    )

    # Run the agent with a simple math question
    result = await Runner.run(math_agent, prompt)

    if not result.final_output:
        raise Exception("No output returned from OpenAI Agents")

    print(f"    Response: {result.final_output}")


async def assert_sentry_captured():
    """Verify Sentry captured the expected spans and data"""
    # Import fixtures validator and setup helper
    # (paths set up by python-test-runner.py wrapper)
    from fixtures import validate_fixture
    from setup import get_mock_sentry_transport

    transport = get_mock_sentry_transport()
    spans = transport.get_spans()
    transactions = transport.get_transactions()
    events = transport.get_events()
    
    print(f"    Captured: {len(spans)} spans, {len(transactions)} transactions, {len(events)} events")



    result = validate_fixture("G1", spans, transactions, events)

    if not result["passed"]:
        print("    ✗ Validation failed:")
        for error in result["errors"]:
            print(f"      - {error}")
        raise Exception(f"Fixture validation failed with {len(result['errors'])} error(s)")

    print("    ✓ All fixture validations passed")


# The main function is exported for the test runner to call
# No need for __main__ block - runner will import and call main()
